ElementX/Sources/Screens/ComposerToolbar/View/VoiceMessageRecordingButton.swift

```diff
@@ -17,79 +17,64 @@
 import Compound
 import SwiftUI
 
+enum VoiceMessageRecordingButtonMode {
+    case idle
+    case recording
+}
+
 struct VoiceMessageRecordingButton: View {
+    let mode: VoiceMessageRecordingButtonMode
     var startRecording: (() -> Void)?
-    var stopRecording: ((_ minimumRecordTimeReached: Bool) -> Void)?
-    
-    @ScaledMetric private var tooltipPointerHeight = 6
+    var stopRecording: (() -> Void)?
     
-    @State private var buttonPressed = false
-    @State private var recordingStartTime: Date?
-    @State private var showTooltip = false
-    @State private var frame: CGRect = .zero
-    
-    private let minimumRecordingDuration = 1.0
-    private let tooltipDuration = 1.0
     private let impactFeedbackGenerator = UIImpactFeedbackGenerator()
-    private let notificationFeedbackGenerator = UINotificationFeedbackGenerator()
+    @ScaledMetric private var recordingImageSize = 16
     
     var body: some View {
-        Button { } label: {
-            CompoundIcon(buttonPressed ? \.micOnSolid : \.micOnOutline)
-                .foregroundColor(.compound.iconSecondary)
-                .padding(EdgeInsets(top: 6, leading: 6, bottom: 6, trailing: 6))
-        }
-        .readFrame($frame, in: .global)
-        .accessibilityLabel(L10n.a11yVoiceMessageRecord)
-        .onLongPressGesture { } onPressingChanged: { isPressing in
-            buttonPressed = isPressing
-            
-            if isPressing {
-                showTooltip = false
-                recordingStartTime = Date.now
-                impactFeedbackGenerator.impactOccurred()
+        Button {
+            impactFeedbackGenerator.impactOccurred()
+            switch mode {
+            case .idle:
                 startRecording?()
-            } else {
-                if let recordingStartTime, Date.now.timeIntervalSince(recordingStartTime) < minimumRecordingDuration {
-                    withElementAnimation {
-                        showTooltip = true
-                    }
-                    notificationFeedbackGenerator.notificationOccurred(.error)
-                    stopRecording?(false)
-                } else {
-                    impactFeedbackGenerator.impactOccurred()
-                    stopRecording?(true)
-                }
+            case .recording:
+                stopRecording?()
             }
-        }
-        .overlay(alignment: .bottomTrailing) {
-            if showTooltip {
-                tooltipView
-                    .offset(y: -frame.height - tooltipPointerHeight)
+        } label: {
+            switch mode {
+            case .idle:
+                CompoundIcon(\.micOnOutline)
+                    .scaledToFit()
+                    .frame(width: recordingImageSize, height: recordingImageSize)
+                    .padding(14)
```

Comment:
```suggestion
                CompoundIcon(\.micOnOutline)
                    .padding(10)
```
Maybe this code does the same?
Remember compound icons are 24x24 by default. Plus 10 padding we are at 44x44.